ToolBox/Polygon/PolygonDataQueueHandler.cs

```diff
@@ -507,105 +508,235 @@ private IEnumerable<Tick> GetCryptoTradeTicks(HistoryRequest request)
-
-                        var utcTime = Time.UnixNanosecondTimeStampToDateTime(row.ExchangeTimestamp);
+                        isException = true;
+                        offsetStart = offsetEnd;
+                    }
 
-                        if (utcTime < start)
+                    if ((response != "") && (isException == false) && (response != null))
+                    {
+                        var obj = JObject.Parse(response);
+                        var objTicks = obj["results"];
+                        if ((objTicks.Type == JTokenType.Null) || (!objTicks.Any()))
                         {
-                            continue;
-                        }
-
-                        start = utcTime.AddMilliseconds(1);
 
-                        if (utcTime > end)
+                            offsetStart = offsetEnd;    // Finish
+                        }
+                        else
                         {
-                            yield break;
+                            // Iterate through each tick and process
+                            foreach (var objTick in objTicks)
+                            {
+                                // Parse the row
+                                var row = objTick.ToObject<EquityTradeTickResponse>();
+
+                                // Converts the results tick time to the Exchange time zone for usage in the Tick Writer LeanData.cs
+                                var utcTime = Time.UnixNanosecondTimeStampToDateTime(row.SipTimestamp);
+                                var time = GetTickTime(request.Symbol, utcTime);
+
+                                // Get the timestamp of the last tick in the results and use for pagination of the API with offset
+                                if (objTicks.Count() == _apiResultsLimit)
+                                {
+                                    offsetStart = objTicks.Last.ToObject<EquityTradeTickResponse>().SipTimestamp;
+                                }
+                                else
+                                {
+                                    offsetStart = offsetEnd;
+                                }
+
+                                // Look for suspicious trades.   In this case, it uses the SIP mapping for updates of high/low otherwise marks as suspicious
+                                bool suspiciousFlag = false;
+                                // Check SIP Mappings for update high/low and mark as suspicious if not update/high low.    This needs more sophistication
+                                if (row.Conditions != null)
+                                {
+                                    suspiciousFlag = !Polygon.PolygonSIPTradeMappings.TradeUpdateHighLow(row.Conditions[0]);
+                                }
+
+                                // Convert Sale Conditions to Hex and concatenate
+                                string saleConditions = "";
```

Comment:
nit - code style, we usually use `var` all around

Reply:
will fix